packages/ibm-products/src/components/Datagrid/Datagrid/addons/stateReducer.js

```diff
@@ -137,6 +149,53 @@ export const stateReducer = (newState, action) => {
         ...newState,
       };
     }
+    case TOGGLE_ON_PAGE_ALL_ROWS_SELECTED: {
+      const { rows, getRowId, indeterminate, isChecked } = action.payload || {};
+      const previousSelectedRowIds = newState.selectedRowIds || {};
+      const previousSelectedRowData = newState.selectedRowData || {};
+
+      const newSelectedRowIds = {};
+      if (rows) {
+        const newSelectedRowData = {};
+        rows.forEach((row) => {
+          const props = row.getRowProps?.();
+          if (props && props.disabled) {
+            return;
+          }
+          if (props) {
+            if (!indeterminate && !isChecked) {
```

Comment:
```suggestion
            if (!indeterminate && isChecked) {
```

Did you mean this instead? Otherwise, the value of `datagridState?.state?.selectedRowData` is always `{}`.

Reply:
right